web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -71,6 +76,13 @@ export interface ConnectMyComputerContext {
   killAgent(): Promise<[void, Error]>;
   isAgentConfiguredAttempt: Attempt<boolean>;
   markAgentAsConfigured(): void;
+  /**
+   * logsFromStartTimeout contains the last 10 lines of logs captured at the time of the start
+   * attempt timing out. Present only if the attempt timed out.
+   *
+   * Prefer to use timeoutLogs from currentAction if possible.
+   */
+  logsFromStartTimeout: string;
```

Comment:
This would be an awesome idea. I went and implemented it, but unfortunately we need logs from the moment that the timeout has occurred, not from the moment the agent has exited.

Instead of seeing logs related to problems with connecting, you're going to see something like this:

```
ERROR REPORT:
Original Error: *trace.CompareFailedError teleport process has shutdown
Stack Trace:
	github.com/gravitational/teleport/lib/service/service.go:2746 github.com/gravitational/teleport/lib/service.(*TeleportProcess).RegisterWithAuthServer.func1
	github.com/gravitational/teleport/lib/service/supervisor.go:550 github.com/gravitational/teleport/lib/service.(*LocalService).Serve
	github.com/gravitational/teleport/lib/service/supervisor.go:281 github.com/gravitational/teleport/lib/service.(*LocalSupervisor).serve.func1
	runtime/asm_arm64.s:1197 runtime.goexit
User Message: teleport process has shutdown] pid:2731.1 service:register.instance service/supervisor.go:286
2023-09-15T16:33:32+02:00 WARN [PROC:1]    Unable to reuse Instance client for Node. (not available) pid:2731.1 service/connect.go:1066
```

Reply:
One thing I liked about this proposed model though is that it makes sense that logs are included only with objects that describe some kind of an exited state. Generally we want to display logs only when something wrong has happened.

I think it might make me pull the trigger and add `error` to `useAsync`, which is what we're missing here – an ability to pass the logs through `joinClusterAttempt` somehow.

It's too bad that having `error` in `useAsync` won't help us utilize the compiler as much – after all, we'll be able to pass arbitrary errors there, without having some kind of an enum of possible error states. But we cannot really circumvent that as this is how JS works.

Reply:
>This would be an awesome idea. I went and implemented it, but unfortunately we need logs from the moment that the timeout has occurred, not from the moment the agent has exited.

😞

>I think it might make me pull the trigger and add error to useAsync, which is what we're missing here – an ability to pass the logs through joinClusterAttempt somehow.

Yeah, I think it will be much better than the `logsFromStartTimeout` property.

Reply:
@gzdunek I'm yet to address the rest of your feedback, but please take a look at the useAsync changes in CMC docs.

I'm going to push a separate PR with the useAsync change, but I'm going to leave a commit with these changes on this branch anyway.

Reply:
The changes to the documents look good to me!